include/crm/common/ipc_attrd.h

```diff
@@ -32,6 +32,21 @@ typedef struct {
     } data;
 } pcmk__attrd_api_reply_t;
 
+/*!
+ * \internal
+ * \brief Get the value of an attribute from pacemaker-attrd
+ *
+ * \param[in] api           Connection to pacemaker-attrd
+ * \param[in] host          Look up the attribute on this host
+ * \param[in] name          Attribute name
+ * \param[in] user_name     ACL user to pass to pacemaker-attrd
+ * \param[in] options       Bitmask of pcmk__node_attr_opts
+ *
+ * \return Standard Pacemaker return code
+ */
+int pcmk__attrd_api_query(pcmk_ipc_api_t *api, const char *host, const char *name,
+                          const char *user_name, int options);
```

Comment:
It doesn't seem too difficult to add versions of all the functions in this header file to handle multiple.  I think the hardest part is making a reasonable command line interface.

Fixed the options type.

Reply:
For the command line interface we have some possibilities:

* Allow multiple name/value options like "attrd_updater -n attr1 -U value1 -n attr2 -U value2" (main question being whether we can handle that easily with GOption, and also it would be pretty easy for the user to get confused and mess up the syntax)
* Change the options to take name=value and allow multiple, like "attrd_updater -U attr1=value1 -U attr2=value2" (easy for GOption, easy to read, and hard to mess up; main problem would be backward compatibility, maybe use the new interpretation only if no -n is given, or deprecate -U and add a new option altogether; attribute names couldn't have = in them or we'd need to support escaping somehow)
* Add an option that takes a list, like "attrd_updater --set attr1=value1,attr2=value2" (we'd have to support escaping = and , in names and values)

For any design, we have to decide whether to allow anything besides name/value to be set differently per attribute in a single command line (node/delay). For simplicity I'd lean to no, use multiple commands if you want different nodes/delays.

... but, the CLI is not actually essential. The feature would primarily be for the daemons (particularly the controller) to use.